### DISCUSSION TITLE
Need help on windows to linux cmd conversion

### DISCUSSION THREAD

**Discussion by SwarnenduGhosh** [ImageMagick/ImageMagick#5261] [Help]
I have a couple of cmd for generating watermark:

`magick convert ${INPUT_FILE} play-tag.png -gravity Center -composite ${OUTPUT_FILE}`

`magick convert ${INPUT_FILE} -gravity SouthEast -fill grey -font Calibri -pointsize 30 -annotate +25+15 ${copyright-value} ${OUTPUT_FILE}`

They work just fine in windows, but in linux they don't. If I try to replace cmd to convert convert also for linux, it doesn't work. 
Please suggest the suitable cmd.

**Comment by fmw42** [comment]
In Imagemagick 7, use magick, not magick convert. In Imagemagick 6, use convert, not magick convert.

**Reply by SwarnenduGhosh** [reply]
Thanks for the quick help!
Would 'magick' work the same way in linux as well for ImageMagick 7?

**Reply by fmw42** [reply]
The same way in Linux as compared to what?  The IM 7 syntax is the same for all systems, except for parentheses in Linux need escaping and % do not need escaping. And the escape character in Linux is \ compared to ^ or % in Windows and the new line character is \ in Linux compared to ^ in Windows.

IM 7 uses magick compared to IM 6 which uses convert.  IM 7 has more new features compared to IM 6, but syntax structure is more strict in IM 7 than it was in IM 6.

**Reply by SwarnenduGhosh** [reply]
Really appreciate your time here. If you see, 
`convert ${INPUT_FILE} -gravity SouthEast -fill grey -font Calibri -pointsize 30 -annotate +25 +15 ${VIDEO_LENGTH} ${OUTPUT_FILE}
`
doesn't have any parenthesis or escape char, the${XYZ} we are replacing in with dynamic value.
It works fine in windows and not in linux.

**Comment by fmw42** [comment]
Use -annotate +25+15.  See https://imagemagick.org/Usage/text/#annotate

Please use the <> button after highlighting your code so that it does not get strangely formatted.

**Reply by SwarnenduGhosh** [reply]
sorry for the printing mistake, I tried both annotate +25 +15 and annotate 25 15 in linux. Neither worked actually. Is the cmd correct in general? Also is annotate, font etc supported in IM 6?

**Comment by snibgo** [comment]
Remove the space between the numbers, like this:
```
-annotate +25+15 HelloWorld
```

**Reply by SwarnenduGhosh** [reply]
I am actually using annotate +25+15 only, (If you see the original), it's somehow still not working.

**Comment by fmw42** [comment]
Try putting your ${copyright-value} in double quotes. Also try doing it with hard-coded file names and text string and not variables. Also use magick, not magick convert.  Does that still not work?  What is the command line you used?
